### DISCUSSION TITLE
Tracking TUI Progress

### DISCUSSION THREAD

**Discussion by shmup** [shmup/awesome-cosmopolitan#2] [TUI]
Here we will track the work being done in this space.

User @ghaerr has a nice start with https://github.com/ghaerr/dflat

Not yet integrated into Cosmo, but planning to in time. The lower-level mouse and arrow key handling are in separate files and are designed to be easily pulled out and used to drive Cosmo programs that want to use ANSI terminal sequence recognition, even if not going full-TUI.

**Comment by ghaerr** [comment]
I've completed the first port of [D-Flat](https://github.com/ghaerr/dflat) to Cosmopolitan. It seems to run well, but I'd like to see how it works on other systems (tested and compiled only on macOS).

Here's a screenshot:
![Screen Shot 2022-09-10 at 3 32 48 PM](https://user-images.githubusercontent.com/11985637/189504059-d105bd17-7288-4172-b3aa-c51a3208ef52.png)

And here's the precompiled APE: 
[dflat.zip](https://github.com/shmup/awesome-cosmopolitan/files/9541496/dflat.zip)

To build your own, pull down D-Flat and edit `make-cosmo` and `cosmocc` to reflect the installation directory of Cosmopolitan and the compiler, then run `make-cosmo`.
